### PR TITLE
Add document to showcase scaleout on hpu

### PR DESCRIPTION
**Pull request by jerome-habana**
Signed-off-by: Jerome <janand@habana.ai>

## What does this PR do?

Fixes #\<issue_number>

## Before submitting

- [ ] Was this **discussed/approved** via a GitHub issue? (not for typos and docs)
- [ ] Did you read the [contributor guideline](https://github.com/Lightning-AI/lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [ ] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [ ] Did you make sure to **update the documentation** with your changes? (if necessary)
- [ ] Did you write any **new necessary tests**? (not for typos and docs)
- [ ] Did you verify new and **existing tests pass** locally with your changes?
- [ ] Did you list all the **breaking changes** introduced by this pull request?
- [ ] Did you **update the CHANGELOG**? (not for typos, docs, test updates, or minor internal changes/refactors)

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing, make sure you have read the [review guidelines](https://github.com/Lightning-AI/lightning/wiki/Review-guidelines). In short, see the following bullet-list:

- [ ] Is this pull request ready for review? (if not, please submit in draft mode)
- [ ] Check that all items from **Before submitting** are resolved
- [ ] Make sure the title is self-explanatory and the description concisely explains the PR
- [ ] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?

Make sure you had fun coding 🙃

### CODE REVIEW

File: src/pytorch_lightning/CHANGELOG.md [hunk 1]
```diff
@@ -15,6 +15,9 @@ The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/).
 - Added support for passing extra init-parameters to the `LightningDataModule.from_datasets` ([#14185](https://github.com/Lightning-AI/lightning/issues/14185))
 
 
+- Added document to showcase scale-out with hpu ([#14357](https://github.com/Lightning-AI/lightning/issues/14357))
+
+
```
Comment (akihironitta) [inline src/pytorch_lightning/CHANGELOG.md:20]: No need to add a changelog entry for changes in docs.
```suggestion
```

File: docs/source-pytorch/accelerators/hpu_basic.rst [hunk 2]
```diff
@@ -47,6 +47,40 @@ It uses :class:`~pytorch_lightning.strategies.hpu_parallel.HPUParallelStrategy`
...
+Scale-out on Gaudis
+-------------------
+
+To train a Lightning model using multiple HPU nodes, set the ``num_nodes`` parameter with the available nodes in the Trainer class
+
+.. code-block:: python
+
+    trainer = Trainer(accelerator="hpu", devices=8, strategy="hpu_parallel", num_nodes=2)
+
+In addition to this, the following environment variables needs to be set to establish communication across nodes. Check out the documentation on ` Cluster <https://pytorch-lightning.readthedocs.io/en/stable/clouds/cluster.html>`__ for more details.
```
Comment (rohitgr7) [inline docs/source-pytorch/accelerators/hpu_basic.rst:59]:
```suggestion
In addition to this, the following environment variables need to be set to establish communication across nodes. Check out the documentation on :doc:`Cluster Environment <../clouds/cluster>` for more details.
```

File: docs/source-pytorch/accelerators/hpu_basic.rst [hunk 3]
```diff
@@ -47,6 +47,40 @@ It uses :class:`~pytorch_lightning.strategies.hpu_parallel.HPUParallelStrategy`
 
 ----
 
+Scale-out on Gaudis
+-------------------
+
+To train a Lightning model using multiple HPU nodes, set the ``num_nodes`` parameter with the available nodes in the Trainer class
```
Comment (rohitgr7) [inline docs/source-pytorch/accelerators/hpu_basic.rst:53]:
```suggestion
To train a Lightning model using multiple HPU nodes, set the ``num_nodes`` parameter with the available nodes in the ``Trainer`` class.
```

File: docs/source-pytorch/accelerators/hpu_basic.rst [hunk 4]
```diff
@@ -47,6 +47,40 @@ It uses :class:`~pytorch_lightning.strategies.hpu_parallel.HPUParallelStrategy`
...
+    trainer = Trainer(accelerator="hpu", devices=8, strategy="hpu_parallel", num_nodes=2)
+
+In addition to this, the following environment variables needs to be set to establish communication across nodes. Check out the documentation on ` Cluster <https://pytorch-lightning.readthedocs.io/en/stable/clouds/cluster.html>`__ for more details.
+
+- *MASTER_PORT* - required; has to be a free port on machine with NODE_RANK 0
+- *MASTER_ADDR* - required (except for NODE_RANK 0); address of NODE_RANK 0 node
+- *WORLD_SIZE* - required; how many workers are in the cluster
+- *NODE_RANK* - required; id of the node in the cluster
+
+The trainer needs to be instantiated on every node that is participating in the training
```
Comment (rohitgr7) [inline docs/source-pytorch/accelerators/hpu_basic.rst:66]:
```suggestion
The trainer needs to be instantiated on every node participating in the training.
```

**Review by rohitgr7 — APPROVED** [review]

File: docs/source-pytorch/accelerators/hpu_basic.rst [hunk 5]
```diff
@@ -47,6 +47,40 @@ It uses :class:`~pytorch_lightning.strategies.hpu_parallel.HPUParallelStrategy`
 
 ----
 
+Scale-out on Gaudis
```
Comment (awaelchli) [inline docs/source-pytorch/accelerators/hpu_basic.rst:50]:
```suggestion
Run across many machines
```
Or something similar? The term "scale-out" is lesser known imo.

**Review by awaelchli — APPROVED** [review]

File: docs/source-pytorch/accelerators/hpu_basic.rst (hunk 5 above)
Comment (jerome-habana) [reply docs/source-pytorch/accelerators/hpu_basic.rst:50]: I've used it for distributed training in hpu docs, so aligning with the same.
